src/character.cpp

```diff
@@ -6817,40 +6817,28 @@ float Character::healing_rate( float at_rest_quality ) const
 float Character::healing_rate_medicine( float at_rest_quality, const body_part bp ) const
 {
     float rate_medicine = 0.0f;
-    float bandaged_rate = 0.0f;
-    float disinfected_rate = 0.0f;
 
-    const effect &e_bandaged = get_effect( effect_bandaged, bp );
-    const effect &e_disinfected = get_effect( effect_disinfected, bp );
+    for( const auto &elem : *effects ) {
```

Comment:
Quite the ugly type, I'm not sure how much it improves versus retracts
```suggestion
    for( const std::pair<const efftype_id, std::unordered_map<body_part, effect, std::hash<int>>>
          &elem : *effects ) {
```